src/librustc/middle/infer/type_variable.rs

```diff
@@ -195,6 +195,15 @@ impl<'tcx> TypeVariableTable<'tcx> {
 
         escaping_types
     }
+
+    pub fn unsolved_variables(&self) -> Vec<ty::TyVid> {
+        self.values.iter().enumerate().filter_map(|(i, value)|
```

Comment:
Total nit, but I personally prefer to line these chained calls on separate lines... even though it's wasteful...